### PR TITLE
models, DAO and web session/filter

### PR DESCRIPTION
**Pull request by AnnaMakarchuk**

### CODE REVIEW

File: src/main/java/org/study/DAO/impl/AdministratorDAOImpl.java [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class AdministratorDAOImpl implements AdministratorDAO {
+
+    private static final Logger LOG = Logger.getLogger(AdministratorDAOImpl.class);
+    private static final Connection INSTANCE = TransactionManager.getInstance();
```
Comment (valeriiafatova) [inline src/main/java/org/study/DAO/impl/AdministratorDAOImpl.java]: advice rename to something like 'connection'

File: src/main/java/org/study/DAO/impl/AdministratorDAOImpl.java [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+            TransactionManager.beginCommit();
+            while (resultSet.next()) {
+                administrator = new Administrator(resultSet);
+                LOG.info("Administrator with login " + id + " is selected");
+            }
+            LOG.info("The transaction was successfully");
+        } catch (SQLException e) {
+            LOG.error("Transaction failed, rollback the transaction", e);
+            TransactionManager.endTransaction();
+        } finally {
```
Comment (valeriiafatova) [inline src/main/java/org/study/DAO/impl/AdministratorDAOImpl.java]: Propose to use try with resources

File: src/main/java/org/study/DAO/impl/AdministratorDAOImpl.java [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+            LOG.info("Committing data here...");
+            TransactionManager.beginCommit();
+            while (resultSet.next()) {
+                administrator = new Administrator(resultSet);
+                LOG.info("Administrator with login " + id + " is selected");
+            }
+            LOG.info("The transaction was successfully");
+        } catch (SQLException e) {
+            LOG.error("Transaction failed, rollback the transaction", e);
+            TransactionManager.endTransaction();
```
Comment (valeriiafatova) [inline src/main/java/org/study/DAO/impl/AdministratorDAOImpl.java]: Why u need transaction here?

File: src/main/java/org/study/DAO/impl/AdministratorDAOImpl.java [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+    public void update(Administrator administrator) {
+        PreparedStatement preparedStatement1 = null;
+        ResultSet resultSet1 = null;
+        PreparedStatement preparedStatement2 = null;
+        Connection con = null;
+        try {
+            con = INSTANCE;
+            preparedStatement1 = con.prepareStatement(ROLE_SQL_GETID);
+            preparedStatement1.setString(1, administrator.getUserRole().name().toLowerCase());
+            resultSet1 = preparedStatement1.executeQuery();
```
Comment (valeriiafatova) [inline src/main/java/org/study/DAO/impl/AdministratorDAOImpl.java]: Advice to review Dao structure. hard to read and understand

File: src/main/java/org/study/DAO/impl/AdministratorDAOImpl.java [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+import org.study.configuration.TransactionManager;
+import org.study.models.Administrator;
+import org.study.DAO.AdministratorDAO;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class AdministratorDAOImpl implements AdministratorDAO {
```
Comment (valeriiafatova) [inline src/main/java/org/study/DAO/impl/AdministratorDAOImpl.java]: you have separate table 'admin'?  review this solution

File: src/main/java/org/study/configuration/JDBCConnection.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.TimeZone;
+
+public class JDBCConnection {
+
+    private static final Logger LOG = Logger.getLogger(JDBCConnection.class);
+
+    private final static String URL = "jdbc:mysql://localhost:3306/cinemaproject?verifyServerCertificate=false" +
+            "&useSSL=false&requireSSL=false&useLegacyDatetimeCode=false&amp&serverTimezone="
```
Comment (valeriiafatova) [inline src/main/java/org/study/configuration/JDBCConnection.java]: are you sure you need all this properties? Propose to add encoding properties

File: src/main/java/org/study/configuration/JDBCConnection.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    private final static String PASSWORD = "root";
+
+    private static BasicDataSource basicDataSource;
+
+    /**
+     * Create a simple datasource of the type: org.apache.commons.dbcp2.BasicDataSource
+     * for connection pool
+     */
+    private static BasicDataSource getBasicDataSource() {
+        if (basicDataSource == null) {
```
Comment (valeriiafatova) [inline src/main/java/org/study/configuration/JDBCConnection.java]: not thread safe

File: src/main/java/org/study/models/Hall.java [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+        this.hallName = hallName;
+        this.maxRow = maxRow;
+        this.maxPlacesInRow = maxPlacesInRow;
+    }
+
+    /**
+     * create Hall with constructor with an input parameter ResultSet*
+     */
+    public Hall(ResultSet resultSet) throws SQLException {
+        this.id = ResultSetTablesData.getIntByName(ResultSetTablesData.ID, resultSet);
```
Comment (valeriiafatova) [inline src/main/java/org/study/models/Hall.java]: logic in Model - not the best solution

File: src/main/java/org/study/services/impl/AdministratorServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+import org.study.factories.DAOFactory;
+import org.study.factories.DAOType;
+import org.study.models.Administrator;
+import org.study.services.AdministratorService;
+
+public class AdministratorServiceImpl implements AdministratorService {
+    /**
+     * create instance of DAO interface with type ADMINISTRATOR
+     */
+    private static final AdministratorDAO INSTANCE =
```
Comment (valeriiafatova) [inline src/main/java/org/study/services/impl/AdministratorServiceImpl.java]: rename at least to dao
Not clear what means Instance when read code, have to scroll and search variable definition

File: src/main/java/org/study/DAO/impl/AdministratorDAOImpl.java (hunk 1 above)
Comment (AnnaMakarchuk) [reply src/main/java/org/study/DAO/impl/AdministratorDAOImpl.java]: Will be fix in next MR

File: src/main/java/org/study/DAO/impl/AdministratorDAOImpl.java (hunk 3 above)
Comment (AnnaMakarchuk) [reply src/main/java/org/study/DAO/impl/AdministratorDAOImpl.java]: Will be fix in next MR

File: src/main/java/org/study/DAO/impl/AdministratorDAOImpl.java (hunk 4 above)
Comment (AnnaMakarchuk) [reply src/main/java/org/study/DAO/impl/AdministratorDAOImpl.java]: Will be fix in next MR

File: src/main/java/org/study/DAO/impl/AdministratorDAOImpl.java (hunk 5 above)
Comment (AnnaMakarchuk) [reply src/main/java/org/study/DAO/impl/AdministratorDAOImpl.java]: Will be fix in next MR

File: src/main/java/org/study/configuration/JDBCConnection.java (hunk 6 above)
Comment (AnnaMakarchuk) [reply src/main/java/org/study/configuration/JDBCConnection.java]: Will be fix in next MR

File: src/main/java/org/study/configuration/JDBCConnection.java (hunk 7 above)
Comment (AnnaMakarchuk) [reply src/main/java/org/study/configuration/JDBCConnection.java]: Will be fix in next MR

File: src/main/java/org/study/models/Hall.java (hunk 8 above)
Comment (AnnaMakarchuk) [reply src/main/java/org/study/models/Hall.java]: Will be fix in next MR

File: src/main/java/org/study/services/impl/AdministratorServiceImpl.java (hunk 9 above)
Comment (AnnaMakarchuk) [reply src/main/java/org/study/services/impl/AdministratorServiceImpl.java]: Will be fix in next MR

**Pull request merged**